tests/posit/connect/test_content.py

```diff
@@ -542,3 +547,84 @@ def test_app_mode_is_other(self):
         # assert
         assert mock_get_content.call_count == 1
         assert mock_patch_content.call_count == 1
+
+
+class TestContentRepository:
+    @property
+    def base_url(self):
+        return "http://connect.example"
+
+    @property
+    def content_guid(self):
+        return "8ce6eaca-60af-4c2f-93a0-f5f3cddf5ee5"
+
+    @property
+    def content_item(self):
+        return ContentItem(self.params, guid=self.content_guid)
+
+    @property
+    def endpoint(self):
+        return f"{self.base_url}/__api__/v1/content/{self.content_guid}/repository"
+
+    @property
+    def get_value(self):
+        return {
+            "repository": "https://github.com/posit-dev/posit-sdk-py/",
+            "branch": "main",
+            "directory": "integration/resources/connect/bundles/example-flask-minimal",
+            "polling": True,
+        }
+
+    @property
+    def patch_branch_value(self):
+        return "testing-main"
+
+    @property
+    def patch_value(self):
+        ret = {**self.get_value}
+        ret.update({"branch": self.patch_branch_value})
+        return ret
+
+    @property
+    def ctx(self):
+        return Context(requests.Session(), Url(self.base_url))
+
+    @property
+    def params(self):
+        return ResourceParameters(self.ctx.session, self.ctx.url)
+
+    def mock_repository_info(self):
+        content_item = self.content_item
+
+        mock_get = responses.get(self.endpoint, json=self.get_value)
+        repository_info = content_item.repository
+
+        assert isinstance(repository_info, ContentItemRepository)
+        assert mock_get.call_count == 1
+
+        return repository_info
+
+    @responses.activate
+    def test_repository_getter_returns_repository(self):
+        # Performs assertions in helper property method
+        self.mock_repository_info()
+
+    @responses.activate
+    def test_repository_update(self):
+        repository_info = self.mock_repository_info()
+
+        mock_patch = responses.patch(self.endpoint, json=self.patch_value)
+        new_repository_info = repository_info.update(branch=self.patch_branch_value)
+        assert mock_patch.call_count == 1
+
+        for key, value in self.patch_value.items():
+            assert new_repository_info[key] == value
+
+    @responses.activate
+    def test_repository_delete(self):
+        repository_info = self.mock_repository_info()
+
+        mock_delete = responses.delete(self.endpoint)
+        repository_info.delete()
+
+        assert mock_delete.call_count == 1
```

Comment:
Ideally, these tests should follow the existing pattern used in the other test. The json should be stored in a file in the __api__ folder that represents a mocked server response.